modules/nw-ovn-kubernetes-migration.adoc

```diff
@@ -0,0 +1,102 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-openshift-sdn.adoc
+
+[id="nw-ovn-kubernetes-migration_{context}"]
+= Migrating to the OVN-Kubernetes network provider
+
+As a cluster administrator, you can upgrade the Pod network provider for your cluster to the OVN-Kubernetes Pod network provider.
+During the migration, you must reboot every node in your cluster.
+
+[IMPORTANT]
+====
+While performing the migration, your cluster is unavailable and workloads may be interrupted.
+Perform the migration only when an interruption in service is acceptable.
+====
+
+.Prerequisites
+
+* Install the OpenShift Command-line Interface (CLI), commonly known as `oc`.
+* Access to the cluster as a user with the `cluster-admin` role.
+* A cluster installed on bare metal infrastructure configured with the OpenShift SDN Pod network provider.
+* The cluster is in a known good state, without any errors.
+
+.Procedure
+
+. To backup the configuration for the cluster network, enter the following command:
++
+----
+$ oc get Network.config.openshift.io cluster -o yaml
+----
+
+. To enable the migration, set an annotation on the Cluster Network Operator configuration object by entering the following command:
++
+----
+$ oc annotate Network.operator.openshift.io cluster \
+  'networkoperator.openshift.io/network-migration'=""
+----
+
+. To change the network provider, enter the following command:
++
+----
+$ oc patch Network.config.openshift.io cluster \
+  --type='merge' --patch '{ "spec": { "networkType": "OVNKubernetes" } }'
+----
+
+. To confirm the migration disabled the OpenShift SDN network provider, enter the following command. It might take several moments for all the OpenShift SDN Pods to stop.
++
+----
+$ oc get pod -n openshift-sdn --watch
+----
+
+. To complete the migration, reboot each node in your cluster. For example, the following bash script reboots each node in your cluster. The script assumes that you can connect to each host by using `ssh` and that you have configured `sudo` to not prompt for a password.
++
+[source,bash]
+----
+#!/bin/bash
+
+for ip in $(oc get nodes  -o jsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}')
+do
+   echo "reboot node $ip"
+   ssh -o StrictHostKeyChecking=no core@$ip sudo shutdown -r -t 3
+done
+----
+
+. Confirm that the migration succeeded:
+
+.. To confirm that the Pod network provider is OVN-Kubernetes, enter the following command:
++
+----
+$ oc describe clusteroperators/network
+----
```

Comment:
Do we need output here to show what the users should look for?

Reply:
@huiran0826, what output is expected from `oc describe clusteroperators/network`? Thanks!

Reply:
@jboxman  It is better to use 'oc get clusteroperators/network' to check the state of the Cluster Network Operator.
 oc get clusteroperators/network
NAME      VERSION                             AVAILABLE   PROGRESSING   DEGRADED   SINCE
network   4.5.0                                        True        False         False      21h

Use another command to check the network type, here is "Network Type:      OVNKubernetes:
oc describe network.config/cluster
Name:         cluster
Namespace:    
Labels:       <none>
Annotations:  <none>
API Version:  config.openshift.io/v1
Kind:         Network
Metadata:
  Creation Timestamp:  2020-06-15T04:03:28Z
  Generation:          2
  Managed Fields:
    API Version:  config.openshift.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:spec:
        .:
        f:clusterNetwork:
        f:externalIP:
          .:
          f:policy:
        f:networkType:
        f:serviceNetwork:
      f:status:
    Manager:      cluster-bootstrap
    Operation:    Update
    Time:         2020-06-15T04:03:28Z
    API Version:  config.openshift.io/v1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        f:clusterNetwork:
        f:clusterNetworkMTU:
        f:networkType:
        f:serviceNetwork:
    Manager:         cluster-network-operator
    Operation:       Update
    Time:            2020-06-15T04:15:00Z
  Resource Version:  3640
  Self Link:         /apis/config.openshift.io/v1/networks/cluster
  UID:               d04fb727-e036-4d03-b87a-f153088f72fd
Spec:
  Cluster Network:
    Cidr:         10.128.0.0/14
    Host Prefix:  23
  External IP:
    Policy:
  Network Type:  OVNKubernetes
  Service Network:
    172.30.0.0/16
Status:
  Cluster Network:
    Cidr:               10.128.0.0/14
    Host Prefix:        23
  Cluster Network MTU:  1400
  Network Type:         OVNKubernetes
  Service Network:
    172.30.0.0/16
Events:  <none>

Reply:
@huiran0826, how about:

`$ oc get network.config/cluster -o jsonpath='{.status.networkType}{"\n"}'`

Reply:
@jboxman lgtm